tests/client/test_redirects.py

```diff
@@ -243,12 +249,29 @@ class MockDispatch(AsyncDispatcher):
 
 
 async def test_body_redirect(backend):
+    """
+    A 308 redirect should preserve the request body.
+    """
     client = AsyncClient(dispatch=MockDispatch(), backend=backend)
     url = "https://example.org/redirect_body"
     data = b"Example request body"
     response = await client.post(url, data=data)
     assert response.url == URL("https://example.org/redirect_body_target")
-    assert response.json() == {"body": "Example request body"}
+    assert response.json()["body"] == "Example request body"
+    assert "content-length" in response.json()["headers"]
+
+
+async def test_no_body_redirect(backend):
+    """
+    A 303 redirect should remove the request body.
+    """
+    client = AsyncClient(dispatch=MockDispatch(), backend=backend)
+    url = "https://example.org/redirect_no_body"
+    data = b"Example request body"
+    response = await client.post(url, data=data)
+    assert response.url == URL("https://example.org/redirect_body_target")
+    assert response.json()["body"] == ""
+    assert "content-length" not in response.json()["headers"]
```

Comment:
Looks like an assert for Transfer-Encoding is missing. :)

Reply:
True yes, tho it’s not present in this case anyways, since it’s not a streaming request.

Reply:
Right! Not sure we should assert it, unless with set stream=True, but then this test is really about the request body being stripped out anyway. I’m good with the current state.